code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -638,35 +543,16 @@
 
 /mob/living/silicon/robot/update_icons()
 	cut_overlays()
+	icon_state = module.cyborg_base_icon
 	if(stat != DEAD && !(paralysis || stunned || weakened || low_power_mode)) //Not dead, not stunned.
-		var/state_name = icon_state //For easy conversion and/or different names
-		switch(icon_state)
-			if("robot")
-				add_overlay("eyes-standard[is_servant_of_ratvar(src) ? "_r" : ""]") //Cyborgs converted by Ratvar have yellow eyes rather than blue
-				state_name = "standard"
-			if("mediborg")
-				add_overlay("eyes-mediborg[is_servant_of_ratvar(src) ? "_r" : ""]")
-			if("toiletbot")
-				add_overlay("eyes-mediborg[is_servant_of_ratvar(src) ? "_r" : ""]")
-				state_name = "mediborg"
-			if("secborg")
-				add_overlay("eyes-secborg[is_servant_of_ratvar(src) ? "_r" : ""]")
-			if("engiborg")
-				add_overlay("eyes-engiborg[is_servant_of_ratvar(src) ? "_r" : ""]")
-			if("janiborg")
-				add_overlay("eyes-janiborg[is_servant_of_ratvar(src) ? "_r" : ""]")
-			if("minerborg","ashborg")
-				add_overlay("eyes-minerborg[is_servant_of_ratvar(src) ? "_r" : ""]")
-				state_name = "minerborg"
-			if("peaceborg")
-				add_overlay("eyes-peaceborg[is_servant_of_ratvar(src) ? "_r" : ""]")
-			if("syndie_bloodhound")
-				add_overlay("eyes-syndie_bloodhound")
-			else
-				add_overlay("eyes")
-				state_name = "serviceborg"
+		if(!eye_lights)
+			eye_lights = new(icon, "[module.special_light_key ? "[module.special_light_key]":"[module.cyborg_base_icon]"]_e[is_servant_of_ratvar(src) ? "_r" : ""]")
+		else
+			eye_lights.icon_state = "[module.special_light_key ? "[module.special_light_key]":"[module.cyborg_base_icon]"]_e[is_servant_of_ratvar(src) ? "_r" : ""]"
+		eye_lights.icon = icon
+		add_overlay(eye_lights)
```

Comment:
While it might seem minor, I must protest having the entire :? operator sequence from hell being copied and pasted twice for no reason. You already have the icon set either way, you may as well extend this treatment to the icon_state as well.